file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+      operationId: ListServiceGroups
+      x-ni-operation: listFiles
+      x-ni-privilege: ViewMetadata
+      x-ni-request-all-privileges: true
+      responses:
+        200:
+          description: OK
+          schema:
+            description: Service group information
+            title: ListServiceGroupsResponse
+            type: object
+            properties:
+              _links:
+                type: object
+                description: >-
+                  Links that apply to the collection of service groups:
+
+                  - self: Link to the collection of service groups
+                properties:
+                  self:
+                    $ref: '#/definitions/Link'
+              serviceGroups:
+                type: array
+                description: Array of service groups within the file service
+                items:
+                  $ref: '#/definitions/ServiceGroup'
+        401:
+          $ref: '#/responses/Unauthorized'
+  /v1/service-groups/Default/files:
+    get:
+      tags: [files]
+      summary: List available files
+      description: >-
+        Lists available files on the Skyline File service.
+
+        The skip and take parameters can be used return paged responses. The
+        user can skip an unsigned integer number of entries, and take an
+        unsigned integer number of entries. A list of 100 files with a skip
+        value of 50 and a take value of 25 will return entries 51 through 75.
+
+        The orderBy and orderByDescending fields are intended to manage sorting
+        the list by metadata objects. The value of the orderBy field should be
+        the name of a metadata key. The elements in the list are sorted
+        ascending by default. If the orderByDescending parameter is specified,
+        the elements in the list are sorted based on it's value. The
+        orderByDescending value must be a boolean string. The elements in the
+        list are sorted ascending if false and descending if true.
+      operationId: ListAvailableFiles_GET
+      x-ni-operation: listFiles
+      x-ni-privilege: ViewMetadata
+      x-ni-request-all-privileges: true
+      parameters:
+        - in: query
+          name: skip
+          description: How many files to skip in the result when paging
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: >-
+            How many files to return in the result, or 0 to use a default
+            defined by the service
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+          maximum: 10000
+        - in: query
+          name: orderBy
+          description: The name of the metadata key to sort by
+          type: string
+          enum:
+            - created
+            - id
+            - size
+        - in: query
+          name: orderByDescending
+          description: Whether to sort descending instead of ascending
```

Comment:
Possible reworking of above description into parameter description here:  Specifies whether to sort in descending order instead of ascending.

[I think the rest of the longer description from above goes without saying given the data type + description for this parameter.]